Cython/Utility/CommonStructures.c

```diff
@@ -116,3 +139,68 @@ static PyTypeObject *__Pyx_FetchCommonTypeFromSpec(PyObject *module, PyType_Spec
     goto done;
 }
 
+//////////////////// CommonTypesMetaclass.module_state_decls ////////////////////
+
+PyTypeObject *__pyx_CommonTypesMetaclassType;
+
+//////////////////// CommonTypesMetaclass.module_state_traverse ///////////////////
+
+Py_VISIT(visit_module_state->__pyx_CommonTypesMetaclassType);
+
+//////////////////// CommonTypesMetaclass.module_state_clear ///////////////////
+
+Py_CLEAR(clear_module_state->__pyx_CommonTypesMetaclassType);
+
+//////////////////// CommonTypesMetaclass.init //////////////////
+//@substitute: naming
+
+if (likely(__pyx_CommonTypesMetaclass_init($module_cname) == 0)); else
+
+/////////////////////////// CommonTypesMetaclass.proto ////////////////////////
+
+static int __pyx_CommonTypesMetaclass_init(PyObject *module); /* proto */
+
+#define __Pyx_CommonTypesMetaclass_USED
+
+//////////////////// CommonTypesMetaclass ////////////////////
+//@requires: FetchCommonType
+//@substitute: naming
+
+PyObject* __pyx_CommonTypesMetaclass_get_module(CYTHON_UNUSED PyObject *self, CYTHON_UNUSED void* context) {
+    return PyUnicode_FromString(__PYX_ABI_MODULE_NAME);
+}
+
+static PyGetSetDef __pyx_CommonTypesMetaclass_getset[] = {
+    {"__module__", __pyx_CommonTypesMetaclass_get_module, NULL, NULL, NULL},
+    {0, 0, 0, 0, 0}
+};
```

Comment:
Can't we just store this as a constant in the type's attributes? (Although it might not be 'just' that simple if we need to add it after the fact, not knowing if it's there already etc.)

Reply:
I don't think so. Mainly because `cython_function_or_method` needs to have a descriptor so an individual CyFunction can store its `__module__`. Which means that `type(cython_function_or_method)` also needs to implement it as a descriptor.

(Of course, one option would be just to put `__module__` in `cython_function_or_method`'s dict, rather than implement it as a descriptor. I didn't particularly like this option because it means always creating a dict for every cyfunction. But it might have been easier)